### ISSUE TITLE
[$250] iOS Hybrid - Lack of debug mode in the settings - Troubleshoot for IOS Hybrid app

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#50843]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 9.0.49-0
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**If this was caught during regression testing, add the test name, ID and link from TestRail:** N/A
**Email or phone of affected tester (no customers):** applausetester+en@applause.expensifail.com
**Issue reported by:** Applause - Internal Team

**Issue found when executing PR** https://github.com/Expensify/App/pull/50519
## Action Performed:
1. Open the Expenisfy Hybrid app
2. Sign into a valid account and access the ND features within the app
3. Go to Account section > Troubleshoot > Scroll down

## Expected Result:
User expects to see the Debug option

## Actual Result:
No Debug option is shown

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [x] iOS: Native
- [ ] iOS: mWeb Safari
- [ ] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

<details>
  <summary>Add any screenshot/video evidence</summary>

  
</details> 

https://github.com/user-attachments/assets/84c7107a-885a-4e09-82e1-997569a3133e

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~021846335027010160243</li>
        <li>Upwork Job ID: 1846335027010160243</li>
        <li>Last Price Increase: 2024-10-29</li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @situchan</details>

**Comment by lanitochka17** [comment]
@greg-schroeder  FYI I haven't added the External label as I wasn't 100% sure about this issue. Please take a look and add the label if you agree it's a bug and can be handled by external contributors

**Comment by Shahidullah-Muffakir** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Debug mode is not displayed in the Troubleshoot page
### What is the root cause of that problem?
we are only displaying Debug mode and other testing options if we are not in the production env:
https://github.com/Expensify/App/blob/26a9fc450e1ece9f258d0309639c61ec94c95aa3/src/pages/settings/Troubleshoot/TroubleshootPage.tsx#L153

### What changes do you think we should make in order to solve the problem?
if we want to show the debug mode in the prod env.
can remove it from the **TestToolMenu**, and directly show it in the TroubleshootPage, without checking for env

we can this :
https://github.com/Expensify/App/blob/26a9fc450e1ece9f258d0309639c61ec94c95aa3/src/components/TestToolMenu.tsx#L49-L56

in the TroubleshootPage , above the TestToolMenu

**Comment by Shahidullah-Muffakir** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Debug mode is not displayed in the Troubleshoot page
### What is the root cause of that problem?
we are only displaying Debug mode and other testing options if we are not in the production env:
https://github.com/Expensify/App/blob/26a9fc450e1ece9f258d0309639c61ec94c95aa3/src/pages/settings/Troubleshoot/TroubleshootPage.tsx#L153

### What changes do you think we should make in order to solve the problem?
if we want to show the debug mode in the prod env.
can remove it from the **TestToolMenu**, and directly show it in the TroubleshootPage, without checking for env

we can add this :
https://github.com/Expensify/App/blob/26a9fc450e1ece9f258d0309639c61ec94c95aa3/src/components/TestToolMenu.tsx#L49-L56

in the TroubleshootPage , above the TestToolMenu

**Comment by situchan** [comment]
can we allow debug mode in hybrid app?
tagging hybrid app experts: @AndrewGable @staszekscp @mateuuszzzzz

**Comment by AndrewGable** [comment]
I believe we can, but let's confirm with @staszekscp and @mateuuszzzzz

**Comment by situchan** [comment]
https://github.com/Expensify/App/blob/04214cdb3e35816fd28a640f1b3c1b23bd7407b0/src/libs/Environment/getEnvironment/index.native.ts#L32-L36
@staszekscp @AndrewGable are we fine to remove this block, which reverts [this PR](https://github.com/Expensify/App/pull/37486)?

**Comment by staszekscp** [comment]
@situchan We'll have to remove this code in order to fix the issue, but I think we'll have to do one more thing on the OldDot side of things to make it work correctly, I'll come back to you with an update

**Comment by staszekscp** [comment]
So, I'm back with an update 😄 On the `Mobile-Expensify` side, we'll have to link a Native Module called `EnvironmentChecker`, which is an easy fix. Combining it with removal of the code mentioned [here](https://github.com/Expensify/App/issues/50843#issuecomment-2428357985) should fix iOS.

Nevertheless we'll have a problem with Android. Currently we base our beta check on this line: https://github.com/Expensify/App/blob/main/src/libs/Environment/betaChecker/index.android.ts#L27. We get the latest release tag, and compare it with the version in `package.json`. If the versions match, we're on production, if the one in `package.json` is higher, we're on a beta build.

The problem is that the HybridApp's production version is different than the release NewDot version. Eg. currently we're on 9.0.47 on HybridApp, and 9.0.51 on NewDot. We cannot simply swap `CONST.GITHUB_RELEASE_URL` value to the `Mobile-Expensify` url, because it's not a public repo, and AFAIK would require additional authentication. Maybe we could create some dummy public repo that would keep that information only for HybridApp version so the only thing we have to do on the NewDot side is to swap a url? Or maybe to have some API endpoint to check the release version?

cc: @Julesssss

**Comment by Julesssss** [comment]
Thanks for the detailed explanation. I wonder why the debug tool is restricted by version instead of email, internal beta, etc... It seems necessarily complex, ad restricts us from debugging prod. Though I suppose with external contributors its better to keep access open rather than locked down...

What if we locked the beta tools behind a hidden menu? For example, 8 taps of the version code could set an Onyx key that unlocks the debug menu. 

One bad alternative might be to query the Google Play Store API and provide the Mobile-Expensify version via a backend command, though I recall the API being pretty bad and undocumented...

**Comment by staszekscp** [comment]
Yep, as far as I know there is no reliable way to check if a specific build is an Android Play Store Beta build 😢 I suppose that's the reason why the whole version checker tool has been implemented. I'd like to keep using this mechanism in order to introduce less changes on the NewDot side - the hidden menu may expose some options to regular users, that we would like to keep unavailable 😄 I'm not familiar with the whole debugging tools code, but we would have to revise it thoroughly first

Maybe we could change the release tag logic on GitHub so it would match only the HybridApp version, since we're going to deprecate NewDot in the nearest future?

**Comment by AndrewGable** [comment]
I think we are planning on having them be in sync, but we need to change some processes internally before we can do so.

**Comment by staszekscp** [comment]
In this case should we wait until it's changed? I suppose we could easily enable debug mode for iOS (since it's based on a TestFlight link, not the version in `package.json`), and have it disabled for Android until we have releases of both repos in sync

**Comment by Julesssss** [comment]
It could be a while though. I would vote for simplify access to the HybridApp debug tools in the meantime.

**Comment by AndrewGable** [comment]
@puneetlath - Do you think we could solve this in a different way than looking at the versioning for the time being? Maybe we can include the debug tools for certain domains  using the beta manager?

**Comment by Julesssss** [comment]
One concern about a beta is that it'll be a barrier that might prevent users/qa from providing logs due to the extra step. Also we'll have to constantly be adding contributors to the beta.

**Comment by puneetlath** [comment]
Perhaps we can just make it available via the four-finger-tap/cmd+d on production. That way there's still a way to get to it on Android. Or if we ever wanted someone to on production for whatever reason.

**Comment by Julesssss** [comment]
I think 4 finger tap or unlocking the menu item after x taps are both good simple solutions. I don't think we should care about users accidentally discovering these tools (granted with the open repo, its pretty easy for users to discover if they really wanted to).

**Comment by AndrewGable** [comment]
> Perhaps we can just make it available via the four-finger-tap/cmd+d on production

I like this 👍

**Comment by puneetlath** [comment]
I agree with that @Julesssss

**Comment by staszekscp** [comment]
Ok, cool! In that case the implementation will be HybridApp agnostic, and it can be done entirely on standalone NewDot. I suppose this kind of work will require some proposal, and general agreement on the solution, so I'll try to find someone on our side to take care of it! 😄

**Comment by Julesssss** [comment]
Sounds good @staszekscp, let's open a new issue rather than modifying this one. Let me know when you find someone to work on the task and I'll create one for you.

**Comment by blazejkustra** [comment]
Hey @Julesssss! I'm going to work on it, tag me on the issue and I'll start tomorrow morning 😄

**Comment by Julesssss** [comment]
Hey @blazejkustra could you comment [here](https://github.com/Expensify/App/issues/51421) and I'll assign you, thanks!

**Comment by greg-schroeder** [comment]
Okay I'm a bit confused on this one - who is going to take this one on? Or are we still waiting for a valid proposal?

**Comment by greg-schroeder** [comment]
Go away melvin

**Comment by Julesssss** [comment]
I think we can close this once [this](https://github.com/Expensify/App/issues/51421) issue is completed (its awaiting testing on prod)

**Comment by greg-schroeder** [comment]
Held til 11-7

**Comment by Julesssss** [comment]
Still held

**Comment by Julesssss** [comment]
The linked issue is closed. We didn't end up needing C+ review here.
